packages/odyssey-react-mui/src/labs/OdysseyPickers/Picker.tsx

```diff
@@ -300,39 +300,53 @@ const Picker: PickerComponentType = <
   const odysseyDesignTokens = useOdysseyDesignTokens();
 
   const customOptionRender = useCallback<
-    (props: HTMLAttributes<HTMLLIElement>, option: OptionType) => ReactNode
+    NonNullable<
+      MuiAutocompleteProps<
+        OptionType,
+        HasMultipleChoices,
+        undefined,
+        IsCustomValueAllowed
+      >["renderOption"]
+    >
   >(
     (muiProps, option) => {
       const hasDescription = "description" in option && option.description;
       const hasMetadata = "metaData" in option && option.metaData;
       const isLabelOnly = !hasMetadata && !hasDescription;
+      const { key, ...restMuiProps } = muiProps;
```

Comment:
To make it closer to other Odyssey names:
```suggestion
      const { key, ...otherMuiProps } = muiProps;
```